platformio.ini

```diff
@@ -378,6 +385,96 @@ extra_scripts =
     ${board:ttgo-t-display.extra_scripts}
     ${common:prog_ota.extra_scripts}
 
+
+; ********************************************************************************
+; TTGO T-Display S3 ESP32 WiFi and Bluetooth Module Development Board - Programming via USB
+; ********************************************************************************
+[env:ttgo-t-display-S3-usb]
+extends = board:ttgo-t-display-S3, common:prog_usb, mode:selected
+platform = espressif32
+build_flags =
+    ${board:ttgo-t-display-S3.build_flags}
+    ${mode:selected.build_flags}
+extra_scripts =
+    ${board:ttgo-t-display-S3.extra_scripts}
+    ${common:prog_usb.extra_scripts}
+
+; ********************************************************************************
+; TTGO T-Display ESP32 WiFi and Bluetooth Module Development Board
+; ********************************************************************************
+[board:ttgo-t-display-S3]
+extends = common:esp32_env, display:ttgo_tdisplay-S3
+board = esp32-s3-devkitc-1
+board_build.partitions = default_8MB.csv
+board_build.filesystem = littlefs
+board_upload.flash_size = "8MB"
+board_upload.maximum_size = 8388608
+build_flags =
+    ${common:esp32_env.build_flags}
+    ${display:ttgo_tdisplay-S3.build_flags}
+    -DBOARD_LILYGO_T_DISPLAY_S3
+lib_deps =
+    ${common:esp32_env.lib_deps_builtin}
+    ${common:esp32_env.lib_deps_external}
+    ${display:ttgo_tdisplay-S3.lib_deps_builtin}
+    ${display:ttgo_tdisplay-S3.lib_deps_external}
+lib_ignore =
+    ${common:esp32_env.lib_ignore_builtin}
+    ${common:esp32_env.lib_ignore_external}
+    ${display:ttgo_tdisplay-S3.lib_ignore_builtin}
+    ${display:ttgo_tdisplay-S3.lib_ignore_external}
+extra_scripts =
+    pre:./scripts/get_git_rev.py
+
+
+
+; ********************************************************************************
+; TTGO T-Display S3
+; https://github.com/Xinyuan-LilyGO/T-Display-S3
+; ********************************************************************************
+[display:ttgo_tdisplay-S3]
+build_flags =
+    -DUSER_SETUP_LOADED=1
+    -DST7789_DRIVER
+    -DTFT_PARALLEL_8_BIT
+    -DCGRAM_OFFSET
+    -DTFT_INVERSION_ON
+    -DTFT_RGB_ORDER=TFT_BGR
+    -DTFT_WIDTH=170
+    -DTFT_HEIGHT=320
+    -DTFT_CS=6
+    -DTFT_DC=7
+    -DTFT_RST=5
+    -DTFT_WR=8
+    -DTFT_RD=9
+    -DTFT_DATA_PIN_OFFSET_EN
+    -DTFT_BL=38
+    -DTFT_D0=39
+    -DTFT_D1=40
+    -DTFT_D2=41
+    -DTFT_D3=42
+    -DTFT_D4=45
+    -DTFT_D5=46
+    -DTFT_D6=47
+    -DTFT_D7=48
+    -DSPI_FREQUENCY=5000000
+    -DSPI_READ_FREQUENCY=2000000
+    -DSPI_TOUCH_FREQUENCY=2500000
+    -DDISABLE_ALL_LIBRARY_WARNINGS
+    -DTFT_MATRIX_WIDTH=32U
+    -DTFT_MATRIX_HEIGHT=8U
+    -DTFT_PIXEL_WIDTH=8
+    -DTFT_PIXEL_HEIGHT=8
+    -DTFT_PIXEL_DISTANCE=1
+    -DTFT_BRIGHTNESS=200
+lib_deps_builtin =
+    HalTtgoTDisplay
+lib_deps_external =
+    https://github.com/nhjschulz/TFT_eSPI
```

Comment:
bodmer/TFT_eSPI is not possible right now?

Reply:
Tried latest version there. It provides tones of warnings during build about shift count larger 31 and resulting image doesnt display anything.  We can move the patched repo also to pixelix. I cloned it basically only as a checkpoint until the bodmer/TFT_eSPI would work with it again.

Reply:
Ok, will move lib in after pull request was merged.